lib/valueflow.cpp

```diff
@@ -3295,6 +3295,12 @@ struct MemberExpressionAnalyzer : SubExpressionAnalyzer {
         : SubExpressionAnalyzer(e, std::move(val), t, s), varname(std::move(varname))
     {}
 
+    bool match(const Token* tok) const override
+    {
+        return SubExpressionAnalyzer::match(tok) ||
+               (Token::simpleMatch(tok->astParent(), ".") && SubExpressionAnalyzer::match(tok->astParent()));
```

Comment:
Why are we matching the token when the parent matches? That seems wrong.

Reply:
Otherwise `x` (in `s.x`, `lifeTok`) and `s` (`expr`) don't match.

Reply:
I see, we probably need to create a different `match` function for this purpose as there are other uses like this that will suffer the same issue.